File: everything/sentimental-readability/readability.py
# TODO
from cs50 import get_string

text = get_string("text: ")

letter = 0
word = 1
sentence = 0
for i in range(len(text)):
    if text[i].isalpha() == True:
        letter += 1
    elif text[i].isspace() == True:
        word += 1
    elif (text[i] == ".") or (text[i] == "?") or (text[i] == "!"):
        sentence += 1
l = (letter / word * 100)
s = (sentence / word * 100)
index = round(0.0588 * l - 0.296 * s - 15.8)
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}" )

File: everything/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("text: ");

    float letter = 0;
    float word = 1;
    float  sentence = 0;
    for(int i = 0, length = strlen(text); i < length; i++)
    {
        if(isupper(text[i]) || islower(text[i]))
        {
            letter += 1;
        }
        else if(isspace(text[i]))
        {
            word += 1;
        }
        else if(text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sentence += 1;
        }
    }
    float l = (letter / word * 100);
    float s = (sentence / word * 100);
    int index = round(0.0588 * l - 0.296 * s - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}